worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/TimeSlidingHub.java

```diff
@@ -108,12 +121,7 @@ public synchronized void add(long currentTimestamp, N newNode) {
       nodeToCombine.combineNode(newNode);
       sumInfo = Pair.of(nodeToCombine, sumInfo.getRight() + nodesToAdd);
 
-      while (_deque.size() > maxQueueSize) {
-        Pair<Long, N> removed = _deque.removeFirst();
-        N nodeToSeparate = sumInfo.getLeft();
-        nodeToSeparate.separateNode(removed.getRight());
-        sumInfo = Pair.of(nodeToSeparate, sumInfo.getRight() - 1);
-      }
+      removeExpiredNodes();
```

Comment:
Also need delete the element when _deque.size() > maxQueueSize?

Reply:
This condition `!_deque.isEmpty() && currentTime - _deque.getFirst().getLeft() >= timeWindowsInMills` and `_deque.size() > maxQueueSize ` are equivalent because one element in _deque is one second.